rust/main/agents/relayer/src/msg/op_queue.rs

```diff
@@ -74,27 +75,55 @@ impl OpQueue {
         // The other consideration is whether to put the channel receiver in the OpQueue or in a dedicated task
         // that also holds an Arc to the Mutex. For simplicity, we'll put it in the OpQueue for now.
         let mut message_retry_requests = vec![];
+
         while let Ok(message_id) = self.retry_rx.lock().await.try_recv() {
-            message_retry_requests.push(message_id);
+            let uuid = message_id.uuid.clone();
+            message_retry_requests.push((
+                message_id,
+                MessageRetryResponse {
+                    uuid,
+                    processed: 0,
+                    matched: 0,
+                },
+            ));
         }
+
         if message_retry_requests.is_empty() {
             return;
         }
+
         let mut queue = self.queue.lock().await;
+        let queue_length = queue.len();
+
         let mut reprioritized_queue: BinaryHeap<_> = queue
             .drain()
             .map(|Reverse(mut op)| {
-                if message_retry_requests.iter().any(|r| r.op_matches(&op)) {
+                let matches = message_retry_requests
+                    .iter()
+                    .any(|(retry_req, _)| retry_req.pattern.op_matches(&op));
+                if matches {
                     info!(
                         operation = %op,
                         queue_label = %self.queue_metrics_label,
                         "Retrying OpQueue operation"
                     );
-                    op.reset_attempts()
+                    op.reset_attempts();
+                }
+                // update retry metrics
+                for (retry_req, req_metric) in message_retry_requests.iter_mut() {
+                    if retry_req.pattern.op_matches(&op) {
```

Comment:
can we avoid doing this here just for incrementing `.matched`, as we have very similar logic a few lines above? ideally shouldn't need to call `op_matches` more than once unnecessarily

Reply:
Yep :+1: